src/_ert_forward_model_runner/job.py

```diff
@@ -18,6 +22,55 @@
     Start,
 )
 
+logger = logging.getLogger(__name__)
+
+
+def killed_by_oom(pids: set[int]) -> bool:
+    """Will try to detect if a process (or any of its descendants) was killed
+    by the Linux OOM-killer.
+
+    Debug information will be logged through the system logger.
+
+    Since pids can be reused, this can in theory give false positives.
+    """
+
+    if sys.platform == "darwin":
+        return False
+
+    try:
+        dmesg_result = run("dmesg", capture_output=True, check=False)
+        if dmesg_result.returncode != 0:
+            logger.warning(
+                "Could not use dmesg to check for OOM kill, "
+                f"returncode {dmesg_result.returncode} and stderr: {dmesg_result.stderr}"
+            )
+            return False
+    except FileNotFoundError:
+        logger.warning(
+            "Could not use dmesg to check for OOM kill, utility not available"
+        )
+        return False
+
+    oom_lines = "".join(
+        [
+            line
+            for line in dmesg_result.stdout.decode(
+                "ascii", errors="ignore"
+            ).splitlines()
+            if "Out of memory:" in line
+        ]
+    )
+
+    for pid in pids:
+        rhel7_message = f"Kill process {pid}"
```

Comment:
suggestion?
```python
kill_pattern = re.compile(f"kill.*? process {pid}")
...
kill_pattern.search(oom_lines)
```

Reply:
I thought about that, but it hides some specific information about which Linux gives which information. I tried reading oom_kill.c in the kernel history to see if I could split it by kernel version, but not trivial.